core/trino-main/src/test/java/io/trino/operator/window/TestAggregateWindowFunction.java

```diff
@@ -595,4 +595,68 @@ public void testSumAllNulls()
                         .row(null, null, null)
                         .build());
     }
+
+    @Test
+    public void testAverageRowsRollingWithNanAndInfinity()
```

Comment:
We need to verify that we're covering the behavior for aggregations over `double`, `long`, and `real` in the presence of `NaN` / `Infinite` inputs, so maybe we should update `AbstractTestAggregationFunction#testSlidingWindow` to assert on the result of `WindowAccumulator#removeInput` and add tests for `NaN` / `Infinity` to relevant subclasses like `TestDoubleAverageAggregation`, `TestDoubleSumAggregation`, and `TestRealAverageAggregation`

Reply:
Done